### PR TITLE
TechPreview for Knative-Kafka-broker docs

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Jira: https://issues.redhat.com/browse/SRVKE-747

Docs for the Knative Kafka broker becoming Tech-Preview

Direct preview link: https://deploy-preview-39408--osdocs.netlify.app/openshift-enterprise/latest/serverless/knative_eventing/serverless-using-brokers#serverless-using-brokers-creating-kafka-brokers

### CODE REVIEW

File: serverless/knative_eventing/serverless-using-brokers.adoc [hunk 1]
```diff
@@ -24,6 +24,18 @@ include::modules/serverless-creating-broker-annotation.adoc[leveloffset=+2]
 include::modules/serverless-creating-broker-labeling.adoc[leveloffset=+2]
 include::modules/serverless-deleting-broker-injection.adoc[leveloffset=+2]
 
+[id="serverless-using-brokers-creating-kafka-brokers"]
+== Creating a Kafka broker
+
+{ServerlessProductName} provides an alternative Kafka-based implementation of the Knative Broker APIs.
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-using-brokers.adoc:30]:
```suggestion
{ServerlessProductName} provides an alternative Kafka-based implementation of the Knative broker APIs.
```

File: modules/serverless-create-kafka-broker-custom-yaml.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+//  * serverless/knative_eventing/serverless-using-brokers.adoc
+
+[id="serverless-create-kafka-broker-custom-yaml_{context}"]
+= Creating a Kafka broker with custom configuration by using YAML
+
+You can create a Kafka broker by using YAML to create the `Kafka`-based broker object.
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-custom-yaml.adoc:8]:
```suggestion
You can create a Kafka broker by using YAML to create the Kafka-based broker.
```

The formatting goes either lowercase general reference, e.g. "broker", or is formatted using backticks if referring to a specific object, and is followed by the word object or resource, e.g. "the `KafkaChannel` resource. Guidelines are available here https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

File: modules/serverless-create-kafka-broker-custom-yaml.adoc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, and the `KnativeKafka` custom resource are installed on your {product-title} cluster.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+
+.Procedure
+
+. Create a `ConfigMap` for configuration as a YAML file:
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-custom-yaml.adoc:17]:
```suggestion
. Create a `ConfigMap` resource for configuration as a YAML file:
```

File: modules/serverless-create-kafka-broker-custom-yaml.adoc [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+kind: ConfigMap
+metadata:
+  name: my-kafka-broker-config
+data:
+  default.topic.partitions: <partitions>
+  default.topic.replication.factor: <replication_factor>
+  bootstrap.servers: <bootstrap_servers>
+----
+
+. Apply the `ConfigMap` YAML file:
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-custom-yaml.adoc:31]: use either "`ConfigMap` resource" or "config map", your call

File: modules/serverless-create-kafka-broker-custom-yaml.adoc [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. Create a `Kafka` broker object, that points to the above configuration, as a YAML file:
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-custom-yaml.adoc:38]:
```suggestion
. Create a Kafka broker as a YAML file that points to the above configuration:
```

File: modules/serverless-create-kafka-broker-custom-yaml.adoc [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    name: my-kafka-broker-config
+    namespace: <namespace>
+----
+
+. Apply the `Kafka`-based broker YAML file:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-custom-yaml.adoc:61]:
```suggestion
. Apply the broker YAML file:
+
[source,terminal]
----
$ oc apply -f <filename>
----
```

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+.Prerequisites
+
+* A username and password for the Kafka cluster.
+* Choose the SASL mechanism to use, for example `PLAIN`, `SCRAM-SHA-256`, or `SCRAM-SHA-512`.
+* If TLS is enabled, you also need the `ca.crt` certificate file for the Kafka cluster.
+
+[NOTE]
+====
+Red Hat recommends to enable TLS in addition to SASL.
+====
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:17]:
```suggestion
[NOTE]
====
Red Hat recommends enabling TLS in addition to SASL.
====
```

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  --from-literal=password="SecretPassword" \
+  --from-literal=user="my-sasl-user"
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `password`, and `sasl.mechanism`. Do not change them.
+====
+
+. Create a `ConfigMap` for configuration as a YAML file:
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:38]:
```suggestion
. Create a config map as a YAML file:
```

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  bootstrap.servers: <bootstrap_servers>
+  auth.secret.ref.name: <kafka_auth_secret>
+----
+
+. Apply the `ConfigMap` YAML file:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:58]:
```suggestion
. Apply the config map:
+
[source,terminal]
----
$ oc apply -f <filename>
----
```

I'll leave it up to you how you want to format these. I'd probably go with "config map" as shown, but there are other suggestions earlier if you want to use `ConfigMap` either.

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+. R
+
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:61]: remove this?

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+. Reference your the `ConfigMap` on your `Kafka` broker:
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:62]:
```suggestion
. Reference the `ConfigMap` object in your Kafka broker configuration:
```

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: my-kafka-broker-config
+    namespace: <namespace>
+----
+
+.Additional resources
+
+* link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[TLS and SASL on Kafka]
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:82]:
```suggestion
* See link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[Encryption and authentication] in the Red Hat AMQ Streams documentation.
```

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-tls-custom-config_{context}"]
+= Configuring TLS authentication for the Kafka Broker
+
+.Prerequisites
+
+* A Kafka cluster CA certificate as a `.pem` file.
+* A Kafka cluster client certificate and key as `.pem` files.
+
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:12]: all the prerequisites for each module should probably have the same bit about installing serverless, eventing, etc.

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+  --from-file=user.crt=certificate.pem \
+  --from-file=user.key=key.pem
+----
++
+[IMPORTANT]
+====
+Use the key names `ca.crt`, `user.crt`, and `user.key`. Do not change them.
+====
+
+. Create a `ConfigMap` for configuration as a YAML file:
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:31]: same comments above for use of `ConfigMap`

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+metadata:
+  name: my-kafka-broker-config
+data:
+  default.topic.partitions: <partitions>
+  default.topic.replication.factor: <replication_factor>
+  bootstrap.servers: <bootstrap_servers>
+  auth.secret.ref.name: <kafka_auth_secret>
+----
+
+. Apply the `ConfigMap` YAML file:
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:46]: see earlier comments

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. R
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:53]: remove?

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+. Reference your the `ConfigMap` on your `Kafka` broker:
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:55]: see earlier comments re formatting

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: my-kafka-broker-config
+    namespace: <namespace>
+----
+
+.Additional resources
+
+* link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[TLS and SASL on Kafka]
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:75]: Rather than duplicate this for different modules, maybe just add an additional resources section to the assembly and add this there?

File: modules/serverless-kafka-broker-tls-default-config.adoc [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+      authSecretName: tls-config
+  channel:
+    enabled: false
+  source:
+    enabled: false
+----
+
+.Additional resources
+
+* link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[TLS and SASL on Kafka]
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-default-config.adoc:89]: same comments as above re prerequisite info, formatting objects like `ConfigMap` and duplicating links, this would be better as a one time link in the assembly to reduce maintenance later.

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 20]
```diff
@@ -39,14 +39,35 @@ spec:
...
+        enabled: true <4>
+        defaultConfig:
+            bootstrapServers: <bootstrap_servers> <5>
+            numPartitions: <num_partitions> <6>
+            replicationFactor: <replication_factor> <7>
 ----
 <1> Enables developers to use the `KafkaChannel` channel type in the cluster.
 <2> A comma-separated list of bootstrap servers from your AMQ Streams cluster.
 <3> Enables developers to use the `KafkaSource` event source type in the cluster.
+<4> Enables developers to use the Knative Kafka `Broker` implementation in the cluster.
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:52]:
```suggestion
<4> Enables developers to use the Knative Kafka broker implementation in the cluster.
```

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 21]
```diff
@@ -39,14 +39,35 @@ spec:
...
+<5> A comma-separated list of bootstrap servers from your AMQ Streams cluster.
+<6> Defines the number of `partitions` of the Kafka Topics, backed by the `Broker` objects. The default is `10`.
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:54]:
```suggestion
<6> Defines the number of partitions of the Kafka topics, backed by the `Broker` objects. The default is `10`.
```

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 22]
```diff
@@ -39,14 +39,35 @@ spec:
...
+<7> Defines the `replication factor` of the Kafka Topics, backed by the `Broker` objects. The default is `3`.
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:55]:
```suggestion
<7> Defines the replication factor of the Kafka topics, backed by the `Broker` objects. The default is `3`. 
```

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 23]
```diff
@@ -39,14 +39,35 @@ spec:
...
 <3> Enables developers to use the `KafkaSource` event source type in the cluster.
+<4> Enables developers to use the Knative Kafka `Broker` implementation in the cluster.
+<5> A comma-separated list of bootstrap servers from your AMQ Streams cluster.
+<6> Defines the number of `partitions` of the Kafka Topics, backed by the `Broker` objects. The default is `10`.
+<7> Defines the `replication factor` of the Kafka Topics, backed by the `Broker` objects. The default is `3`. 
+
+[NOTE]
+====
+The value provided for `replicationFactor` must match the number of nodes of your AMQ Streams cluster.
+====
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:60]:
```suggestion
[NOTE]
====
The `replicationFactor` value must match the number of nodes of your Red Hat AMQ Streams cluster.
====
```

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 24]
```diff
@@ -39,14 +39,35 @@ spec:
...
+        defaultConfig:
+            bootstrapServers: <bootstrap_servers> <5>
+            numPartitions: <num_partitions> <6>
+            replicationFactor: <replication_factor> <7>
 ----
 <1> Enables developers to use the `KafkaChannel` channel type in the cluster.
 <2> A comma-separated list of bootstrap servers from your AMQ Streams cluster.
 <3> Enables developers to use the `KafkaSource` event source type in the cluster.
+<4> Enables developers to use the Knative Kafka `Broker` implementation in the cluster.
+<5> A comma-separated list of bootstrap servers from your AMQ Streams cluster.
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:53]:
```suggestion
<5> A comma-separated list of bootstrap servers from your Red Hat AMQ Streams cluster.
```

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 25]
```diff
@@ -39,14 +39,35 @@ spec:
...
+====
+
 
 // Install Kafka
 include::modules/serverless-install-kafka-odc.adoc[leveloffset=+2]
 
+[id="serverless-kafka-broker-link"]
+== Using Kafka Broker
+
+Create a xref:../../serverless/knative_eventing/serverless-using-brokers.adoc#serverless-using-brokers-creating-kafka-brokers[Kafka Broker].
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:69]:
```suggestion
Create a xref:../../serverless/knative_eventing/serverless-using-brokers.adoc#serverless-using-brokers-creating-kafka-brokers[Kafka broker].
```

File: modules/serverless-kafka-broker-sasl-default-config.adoc [hunk 26]
```diff
@@ -0,0 +1,96 @@
...
+      authSecretName: scram-user
+  channel:
+    enabled: false
+  source:
+    enabled: false
+----
+
+.Additional resources
+
+* link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.5/html-single/using_amq_streams_on_rhel/index#assembly-kafka-encryption-and-authentication-str[TLS and SASL on Kafka]
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-default-config.adoc:96]: See comments on other sections re links and prereqs

File: modules/serverless-create-kafka-broker-default-yaml.adoc [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+  name: example-kafka-broker
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: kafka-broker-config
+    namespace: knative-eventing
+----
+
+. Apply the `Kafka`-based broker YAML file:
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-default-yaml.adoc:35]:
```suggestion
. Apply the Kafka-based broker YAML file:
```

File: modules/serverless-create-kafka-broker-default-yaml.adoc [hunk 28]
```diff
@@ -0,0 +1,40 @@
...
+You can create a Kafka broker by using YAML with the cluster default configuration, provided in `KnativeKafka` type.
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, and the `KnativeKafka` custom resource are installed on your {product-title} cluster.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+
+.Procedure
+
+. Create a `Kafka`-based broker object as a YAML file:
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-default-yaml.adoc:17]:
```suggestion
. Create a Kafka-based broker as a YAML file:
```

File: modules/serverless-create-kafka-broker-default-yaml.adoc [hunk 29]
```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+//  * serverless/knative_eventing/serverless-using-brokers.adoc
+
+[id="serverless-create-kafka-broker-default-yaml_{context}"]
+= Creating a Kafka broker with default configuration by using YAML
+
+You can create a Kafka broker by using YAML with the cluster default configuration, provided in `KnativeKafka` type.
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-default-yaml.adoc:8]:
```suggestion
You can create a Kafka broker by using YAML with the cluster default configuration.
```

File: modules/serverless-create-kafka-broker-default-yaml.adoc [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, and the `KnativeKafka` custom resource are installed on your {product-title} cluster.
+* You have created a project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
```
Comment (abrennan89) [inline modules/serverless-create-kafka-broker-default-yaml.adoc:13]: should be prereqs for all these modules I think

**Review by abrennan89 — CHANGES_REQUESTED** [review]
@matzew good so far, added some initial comments from first review

**Comment by abrennan89** [comment]
You will need to add the tech preview module somewhere in this doc; let's get the content right and add that at the end since it's just including a module and a product name var.

File: modules/serverless-kafka-broker-tls-custom-config.adoc (hunk 13 above)
Comment (matzew) [reply modules/serverless-kafka-broker-tls-custom-config.adoc:12]: I copied the `Prerequisites` from the `modules/serverless-kafka-tls.adoc` file, which also does not list serverless, eventing etc as required.

File: modules/serverless-kafka-broker-sasl-default-config.adoc (hunk 26 above)
Comment (matzew) [reply modules/serverless-kafka-broker-sasl-default-config.adoc:96]: Both `.Prerequisites` and `.Additional resources` are copied from the `modules/serverless-kafka-sasl.adoc` file, as I used that as inspiration for this subject

File: modules/serverless-kafka-broker-tls-custom-config.adoc (hunk 18 above)
Comment (matzew) [reply modules/serverless-kafka-broker-tls-custom-config.adoc:75]: It's basically the same as in the `modules/serverless-kafka-tls.adoc` file, which I used as inspiration.

File: modules/serverless-kafka-broker-tls-default-config.adoc (hunk 19 above)
Comment (matzew) [reply modules/serverless-kafka-broker-tls-default-config.adoc:89]: Same as in `modules/serverless-kafka-tls.adoc` file, as stated above

**Comment by matzew** [comment]
Thanks a lot for the review, @abrennan89 :hugs: 

I have a few questions to your review comments:
* About the "tech preview module" inclusion. Do you have a sample on what's actually required?
* About the `Prerequisites` and `Additional links`, I can align them, and all the information on each module. Let me know 

Thanks again for the review!

**Comment by abrennan89** [comment]
You can see what's required for tech preview here: https://github.com/openshift/openshift-docs/pull/29571/files#diff-3aa74b3ded76160721560b201cc0564e3f543d9b0533e0368614673c527b694dR9-R10

Basically somewhere in the beginning of the assembly you should include something like:

```
:FeatureName: Whatever the feature name is
include::modules/technology-preview.adoc[leveloffset=+0]
```

**Comment by abrennan89** [comment]
> Prerequisites and Additional links

If you could align the prerequisites and make the additional links just one bullet in the assembly instead of repeating in each module that would be ideal.

Re having an additional resources section in the assembly, see https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

**Comment by matzew** [comment]
> Basically somewhere in the beginning of the assembly you should include something like:
```
:FeatureName: Whatever the feature name is
include::modules/technology-preview.adoc[leveloffset=+0]
```

Thanks! I've added that now, see: https://github.com/openshift/openshift-docs/pull/39408/commits/76d2738e12a14d7a6e7f57c3c4ee5583f6e2db40

**Comment by matzew** [comment]
> Re having an additional resources section in the assembly, see https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

I will try this next!

Thanks a lot for all the useful feedback, @abrennan89 :hugs:

**Comment by abrennan89** [comment]
Thanks @matzew - this looks great so far! Let me know when you're ready for another review.

**Comment by matzew** [comment]
@abrennan89 I've aligned the `.Prerequisites` now.

Regarding the `assembly` I am a bit struggling. Do you have an example that we have in serverless ? 

I was wondering if that is good enough to do _afterwards_. Since we touch some common TLS/SASL configuration references from Strimzi/AMQ-Streams. Those are also relevant for the "KafkaChannel". But not really sure mixing the "refactoring" of that for Broker/Channel with this new document.

Besides the assmbly work, I think my text here is ready

**Comment by abrennan89** [comment]
Yeah I think that's OK for now. Can you please open a Jira to follow up on it so I don't forget? 🙂 
The next step here would be to ask for a QE review and get their sign off.

File: modules/serverless-kafka-broker-sasl-default-config.adoc [hunk 31]
```diff
@@ -0,0 +1,98 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-sasl-default-config_{context}"]
+= Configuring SASL authentication for the Kafka Broker
```
Comment (mgencur) [inline modules/serverless-kafka-broker-sasl-default-config.adoc:6]: This one has the exactly same title as https://github.com/openshift/openshift-docs/pull/39408/files#diff-741342931b1ca442259b80ba2cf945e18bebb1ff9c2c3535e6067ac10c399eb3R6
Perhaps we should differentiate them? Is this current section related to "default" configuration?

File: modules/serverless-kafka-broker-tls-default-config.adoc [hunk 32]
```diff
@@ -0,0 +1,91 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-tls-default-config_{context}"]
+= Configuring TLS authentication for the Kafka Broker
```
Comment (mgencur) [inline modules/serverless-kafka-broker-tls-default-config.adoc:6]: And this one has the exact same title as https://github.com/openshift/openshift-docs/pull/39408/files#diff-a861e7b5c742b621d6f5762a7b8d003a89d1d935de452e54dd6d8c1dd10fc282R6
Let's make it clear how they differ in the title.

**Review by mgencur — COMMENTED** [review]
Hey ! Just a few nits. Mostly about titles.

File: modules/serverless-kafka-broker-sasl-custom-config.adoc [hunk 33]
```diff
@@ -0,0 +1,82 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-sasl-custom-config_{context}"]
+= Configuring SASL authentication for the Kafka Broker
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-sasl-custom-config.adoc:6]:
```suggestion
= Configuring SASL authentication for the Kafka broker
```

File: modules/serverless-kafka-broker-sasl-default-config.adoc (hunk 31 above)
Comment (abrennan89) [reply modules/serverless-kafka-broker-sasl-default-config.adoc:6]: Also, use "broker" not "Broker" in the titles

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 34]
```diff
@@ -0,0 +1,75 @@
+// Module is included in the following assemblies:
+//
+// * serverless/knative_eventing/serverless-kafka.adoc
+
+[id="serverless-kafka-broker-tls-custom-config_{context}"]
+= Configuring TLS authentication for the Kafka Broker
```
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:6]:
```suggestion
= Configuring TLS authentication for the Kafka broker
```

File: modules/serverless-kafka-broker-tls-default-config.adoc (hunk 32 above)
Comment (abrennan89) [inline modules/serverless-kafka-broker-tls-default-config.adoc:6]:
```suggestion
= Configuring TLS authentication for the Kafka broker
```

File: serverless/knative_eventing/serverless-kafka.adoc [hunk 35]
```diff
@@ -39,14 +39,35 @@ spec:
...
+====
+The `replicationFactor` value must match the number of nodes of your Red Hat AMQ Streams cluster.
+====
+
 
 // Install Kafka
 include::modules/serverless-install-kafka-odc.adoc[leveloffset=+2]
 
+[id="serverless-kafka-broker-link"]
+== Using Kafka Broker
```
Comment (abrennan89) [inline serverless/knative_eventing/serverless-kafka.adoc:67]:
```suggestion
== Using a Kafka broker
```

**Review by abrennan89 — CHANGES_REQUESTED** [review]
Some nits in titles

**Comment by matzew** [comment]
@abrennan89 thanks! I've update the suggested code snippets :smile:

**Comment by matzew** [comment]
@mgencur I have updated the titles, as per here: 

https://github.com/openshift/openshift-docs/pull/39408/commits/f522bc90be5c0fea276242c3d135f148b810725b

Thanks a lot!

**Comment by mgencur** [comment]
Thanks, @matzew . Looks good to me now.

**Comment by abrennan89** [comment]
Thanks @matzew - I'll go and ask for a peer review for this now. Please also squash your commits down to 1 commit. Thanks

**Review by abrennan89 — APPROVED** [review]

**Comment by lpettyjo** [comment]
I notice that there are multiple commits. This should be squashed to one.

**Comment by lpettyjo** [comment]
Is there an epic or BZ for this feature?

**Comment by abrennan89** [comment]
>Is there an epic or BZ for this feature?

There's a Jira attached already in the comments. Serverless docs always use Jira unless it's a BZ reported through OpenShift for us, which isn't too common.

File: modules/serverless-create-kafka-broker-default-yaml.adoc [hunk 36]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+//  * serverless/knative_eventing/serverless-using-brokers.adoc
+
+[id="serverless-create-kafka-broker-default-yaml_{context}"]
+= Creating a Kafka broker with default configuration by using YAML
+
+You can create a Kafka broker by using YAML with the cluster default configuration, which is stored in the `kafka-broker-config` config map located in the `knative-eventing` namespace.
+
+:FeatureName: Creating a Kafka broker with the cluster default configuration
```
Comment (lpettyjo) [inline modules/serverless-create-kafka-broker-default-yaml.adoc:10]: Same comment as before regarding module within module.

File: modules/serverless-kafka-broker-sasl-default-config.adoc [hunk 37]
```diff
@@ -0,0 +1,98 @@
...
+    enabled: true
+    defaultConfig:
+      bootstrapServers: <bootstrap_servers>
+      numPartitions: <num_partitions>
+      replicationFactor: <replication_factor>
+      authSecretName: <kafka_auth_secret>
+  channel:
+    enabled: false
+  source:
+    enabled: false
```
Comment (lpettyjo) [inline modules/serverless-kafka-broker-sasl-default-config.adoc:67]: Call out the secret and namespace in this example.

File: modules/serverless-kafka-broker-sasl-default-config.adoc [hunk 38]
```diff
@@ -0,0 +1,98 @@
...
+spec:
+  broker:
+    enabled: true
+    defaultConfig:
+      bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9093
+      authSecretName: scram-user
+  channel:
+    enabled: false
+  source:
+    enabled: false
```
Comment (lpettyjo) [inline modules/serverless-kafka-broker-sasl-default-config.adoc:93]: See preceding comment.

File: modules/serverless-kafka-broker-tls-default-config.adoc [hunk 39]
```diff
@@ -0,0 +1,91 @@
...
+    enabled: true
+    defaultConfig:
+      bootstrapServers: <bootstrap_servers>
+      numPartitions: <num_partitions>
+      replicationFactor: <replication_factor>
+      authSecretName: <kafka_auth_secret>
+  channel:
+    enabled: false
+  source:
+    enabled: false
```
Comment (lpettyjo) [inline modules/serverless-kafka-broker-tls-default-config.adoc:60]: See preceding comment about callouts.

File: modules/serverless-kafka-broker-tls-default-config.adoc [hunk 40]
```diff
@@ -0,0 +1,91 @@
...
+  broker:
+    enabled: true
+    defaultConfig:
+      bootstrapServers: eventing-kafka-bootstrap.kafka.svc:9093
+      authSecretName: tls-config
+  channel:
+    enabled: false
+  source:
+    enabled: false
+----
```
Comment (lpettyjo) [inline modules/serverless-kafka-broker-tls-default-config.adoc:87]: See preceding note about callouts.

File: modules/serverless-kafka-broker-tls-custom-config.adoc [hunk 41]
```diff
@@ -0,0 +1,75 @@
...
+    eventing.knative.dev/broker.class: Kafka
+  name: my-demo-kafka-broker
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: my-kafka-broker-config
+    namespace: <namespace>
+----
+
```
Comment (lpettyjo) [inline modules/serverless-kafka-broker-tls-custom-config.adoc:72]: See preceding comment about callouts.

File: modules/serverless-create-kafka-broker-custom-yaml.adoc [hunk 42]
```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+//  * serverless/knative_eventing/serverless-using-brokers.adoc
+
+[id="serverless-create-kafka-broker-custom-yaml_{context}"]
+= Creating a Kafka broker with custom configuration by using YAML
+
+You can create a Kafka broker by using YAML with custom configuration.
+
+:FeatureName: Creating a Kafka broker with custom configuration
```
[the remark this answers is not in the document]
Comment (matzew) [reply modules/serverless-create-kafka-broker-custom-yaml.adoc:10]: I am not 100% I do understand the suggestion :see_no_evil: 

Is my doc wrong "package"?

File: modules/serverless-kafka-broker-sasl-default-config.adoc (hunk 37 above)
Comment (matzew) [reply modules/serverless-kafka-broker-sasl-default-config.adoc:67]: @lpettyjo with "call out", do you mean adding numbers to them, and providing some description below the code snippet, like here: https://github.com/openshift/openshift-docs/pull/39408/files#diff-e37a5088c4981ab85e59a970a409fc334c60bf2a047d71c8065dbc14a832da03R52-R55  ?

Comment (abrennan89) [reply modules/serverless-kafka-broker-sasl-default-config.adoc:67]: I'm not sure that this is necessarily required - we don't use callouts in a lot of other sample YAML too, and I think it's OK. I would recommend though spelling out what the variables are, e.g. rather than `<num_partitions>` maybe use `<number_of_partitions>` if it doesn't change the meaning, instead of `<kafka_auth_secret>` use `<kafka_authentication_secret>`, etc. Using better described variables in a lot of our other content is maybe the reason callouts aren't always required.

Comment (abrennan89) [reply modules/serverless-kafka-broker-sasl-default-config.adoc:67]: @lpettyjo can you clarify whether the callouts Matthias is describing is what you mean? I've been assuming that's the case but maybe you meant something else? Otherwise yes @matzew , the only callouts I'm aware of are those.

File: modules/serverless-create-kafka-broker-custom-yaml.adoc (hunk 42 above)
[the remark this answers is not in the document]
Comment (abrennan89) [reply modules/serverless-create-kafka-broker-custom-yaml.adoc:10]: @matzew basically you cannot use an `include` here to add the technology preview bit because this file is already a module, i.e. ` modules/serverless-create-kafka-broker-custom-yaml.adoc`
The comment you have included at the top of the file:

```
// Module included in the following assemblies:
//
//  * serverless/knative_eventing/serverless-using-brokers.adoc
```

refers to this. The content here https://github.com/openshift/openshift-docs/pull/39408/files#diff-f77d942aa26ea42173a302709c69c3bde6c8f81aa2d202a8435b717965f92330R27-R37 is including files in an assembly, as an example.

Honestly I'm not sure about how this has been structured in https://github.com/openshift/openshift-docs/pull/39408/files#diff-f77d942aa26ea42173a302709c69c3bde6c8f81aa2d202a8435b717965f92330R27-R37 - this is why I would prefer that engineering just provide content, and then PRs are raised by docs folks who are aware of all the formatting and structure requirements and limitations. It's sort of making more work for us if we have to figure out how to make this PR work than it would be if the work was just scoped properly for the docs team for a release.

[the remark this answers is not in the document]
Comment (matzew) [reply modules/serverless-create-kafka-broker-custom-yaml.adoc:10]: OK, thanks! Let me see tomorrow howto re-structure this a bit differently. 

Otherwise we can close the PR and use it as a reference for content provided by engineering, for a more proper docs PR.

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.11` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.10. And any PR going into `main` must also target the latest version branch (enterprise-4.11).

If the update in your PR does NOT apply to version 4.11 onward, please retarget this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.
